src/test/java/com/redhat/devtools/lsp4ij/features/formatting/JavaClientSideFormatOnCloseBraceTest.java

```diff
@@ -0,0 +1,446 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.redhat.devtools.lsp4ij.client.features.LSPFormattingFeature.FormattingScope;
+import com.redhat.devtools.lsp4ij.fixtures.LSPClientSideOnTypeFormattingFixtureTestCase;
+
+/**
+ * Java-based client-side on-type formatting tests for format-on-close brace. Note that the jdtls range formatter does
+ * not work well enough for format-on-statement terminator or format-on-completion trigger, so this is the only test
+ * for Java/jdtls.
+ */
+public class JavaClientSideFormatOnCloseBraceTest extends LSPClientSideOnTypeFormattingFixtureTestCase {
```

Comment:
Added a new unit test for client-side format-on-close brace handling in Java/jdtls.